internal/cmd/config.go

```diff
@@ -2124,16 +2119,14 @@ func (c *Config) persistentPreRunRootE(cmd *cobra.Command, args []string) error
 		"WINDOWS_VERSION": templateData.windowsVersion,
 	} {
 		for key, value := range group {
-			upperSnakeCaseKey := camelCaseToUpperSnakeCase(key)
+			key := "CHEZMOI_" + groupKey + "_" + camelCaseToUpperSnakeCase(key)
 			valueStr := fmt.Sprintf("%s", value)
-			scriptEnv = append(scriptEnv, "CHEZMOI_"+groupKey+"_"+upperSnakeCaseKey+"="+valueStr)
+			os.Setenv(key, valueStr)
 		}
 	}
 	for key, value := range c.ScriptEnv {
-		scriptEnv = append(scriptEnv, key+"="+value)
+		os.Setenv(key, value)
```

Comment:
We should probably prohibit or at least warn if someone tries to set `CHEZMOI` or `CHEZMOI_*` variables using `scriptEnv`.

Reply:
Good idea! Implementing...

Reply:
Implemented in https://github.com/twpayne/chezmoi/pull/3290/commits/8d0826a0c9baaeb254b5c51cfd5161c3f3e00aa7.

Reply:
I think that (a) warning and (b) ignoring such changes is the best choice, but a `panic(…)` would be just as good from my perspective, when you get to it.